src/Stratis.SmartContracts.CLR/CallDataSerializer.cs

```diff
@@ -63,8 +63,9 @@ protected virtual Result<ContractTxData> SerializeCreateContract(byte[] smartCon
 
             var contractExecutionCode = this.primitiveSerializer.Deserialize<byte[]>(decodedParams[0]);
             object[] methodParameters = this.DeserializeMethodParameters(decodedParams[1]);
+            string[] signatures = (decodedParams.Count > 2) ? this.DeserializeSignatures(decodedParams[2]) : null;
```

Comment:
So the idea here is that if we don't have more than 2 signatures then we don't include it in ContractTxData?

Reply:
Perhaps we should report this back to the use somehow... WDYT?

Reply:
Ah the 2 is checking the number of "sections". First section is execution code. Second section is method parameters. Finally if there is a 3rd section it would be the signatures.

Reply:
Ah of course.  Doh!